ci_framework/roles/override_op_image/tasks/main.yml

```diff
@@ -0,0 +1,52 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Ensure directory exists
+  ansible.builtin.file:
+    path: "{{ cifmw_override_op_image_dir }}/{{ item }}"
+    state: directory
+  loop:
+    - artifacts/manifests/openstack
+    - logs
+
+- name: Update CSVs with custom images
+  when:
+  - cifmw_openstack_svc_image is defined
+  - cifmw_openshift_kubeconfig is defined
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  block:
+    - name: Dump OpenStackControlPlane CR into a file
```

Comment:
At this point, if this role is always called from edpm_prepare, you already have the controlplane CR loaded in a fact, as a result of the kustomization, and, at this point, is already deployed, why making to calls to fetch something that's already there?